### ISSUE TITLE
Audit automated tests to understand scope of disabled tests and what may be required to enable them.

### ISSUE THREAD

**Issue by mrglavas** [OpenLiberty/liberty-tools-intellij#642]
Several automated tests have been disabled for various reasons, creating a gap in our coverage of language server features and other capabilities of the Liberty Tools plug-in for IntelliJ. The goal of this work item is to review the disabled tests, to understand the scope of test coverage that's been excluded from automation and explore next steps for re-enabling those tests.

**Comment by mrglavas** [comment]
### Details of disabled language server tests

=========

**io.openliberty.tools.intellij.it.SingleModMPLSTestCommon.testInsertCodeSnippetIntoJavaPart()**

Tests MicroProfile Language Server code snippet support (specifically the `mpliveness` snippet) in a Java source file. Disabled due to https://github.com/OpenLiberty/liberty-tools-intellij/issues/401 which was an issue with code snippets being offered in invalid contexts. With the Java cursor context improvements this should be working with the latest version of LSP4MP and so we should be able to enable this test once the LSP4iJ integration/development branch is merged into main.

=========

**io.openliberty.tools.intellij.it.SingleModMPLSTestCommon.testMPQuickFixInJavaFile()**

Tests MicroProfile Language Server quick fix support in a Java source file. Specifcally the test deletes a `@Liveness` annotation from the source file which generates a diagnostic for the missing annotation and a quick fix for inserting the annotation. The test verifies that a diagnostic is produced and then selects and applies the quick fix, validating that the annotation has been restored in the source. This test was disabled due to intermittent failures. Previous investigation revealed that the quick fix selection link was intermittently failing to appear on the diagnostic popup panel which seems to be framework related.

=========

**io.openliberty.tools.intellij.it.SingleModMPLSTestCommon.testQuickFixInMicroProfileConfigProperties()**

Tests MicroProfile Language Server quick-fix support for microprofile config entries in the microprofile-config.properties file. Similarly to testMPQuickFixInJavaFile(), this test was disabled due to intermittent failures. Previous investigation revealed that the quick fix selection link was intermittently failing to appear on the diagnostic popup panel which seems to be framework related. If we can find a solution which would allow testMPQuickFixInJavaFile() to run reliably, hopefully that same solution would apply to this test as well.

=========

**io.openliberty.tools.intellij.it.SingleModLibertyLSTestCommon.testQuickFixInServerXML()**

Tests liberty-ls support in server.xml for diagnostics and quickfixes. Disabled for same reason (intermittent failures) as other disabled quickfix tests.

=========

**io.openliberty.tools.intellij.it.SingleModJakartaLSTestCommon.testJakartaQuickFixInJavaPart()**

Tests Jakarta Language Server quick fix support in a Java source file. Applies a quickfix which makes a private method public. This test would have been disabled because the quickfix being tested was not exposed in the original release of Liberty Tools for IntelliJ. With Jakarta quickfixes being delivered on to main branch this test should be runnable, but would expect it also suffers from the same issue(s) that's causing intermittent failures for all the other automated UI quickfix tests.

**Comment by mrglavas** [comment]
In addition to the above, all language server tests are currently disabled on Windows. Four tests which start Liberty in a container are only enabled on Linux because, on other platforms, the docker build process driven by the Liberty Maven/Gradle plugins takes longer than ten minutes. Ten minutes is the timeout set by the plugins and there is currently no way to extend this timeout through the Liberty Tools plugin(i.e. set dockerBuildTimeout).

**Comment by mrglavas** [comment]
Observations / Conclusions / Actions

1. The unit tests for Jakarta provide the only automated test coverage for quick fixes. These are not tested through the language server which means MP and LCLS quick fixes are not exercised by any of the enabled automated tests. All of those tests seem to have been disabled with the same root cause. Determine whether this issue can be addressed to make the tests reliable. 
2. The enabled tests do not cover MP completion because we were not releasing the plug-in with MP completion support. With an update to the latest MP client the disabled test should be enabled.
3. Automated testing of the language servers is limited to Mac and Linux only. Revisit why the tests are not run on Windows and explore if there's a path forward for automation on Windows.
4. Container testing is restricted to Linux and based on our previous investigations seems like that will continue to be a limitation unless the Liberty Maven/Gradle plugins become more flexible with timeouts.

**Comment by mrglavas** [comment]
Opened the following issues to cover the follow-up actions outlined in this issue: https://github.com/OpenLiberty/liberty-tools-intellij/issues/653, https://github.com/OpenLiberty/liberty-tools-intellij/issues/654, https://github.com/OpenLiberty/liberty-tools-intellij/issues/655, https://github.com/OpenLiberty/liberty-tools-intellij/issues/656.
